sdk/storage/storage-blob/src/BlobBatch.ts

```diff
@@ -370,29 +368,23 @@ class InnerBatchRequest {
   }
 
   /**
-   * Create pipeline to assemble sub requests. The idea here is to use exising
-   * credential and serialization/deserialization components, with additional policies to
+   * Create pipelineOptions to assemble sub requests. The idea here is to extend the
+   * default pipeline policies with additional policies to
    * filter unnecessary headers, assemble sub requests into request's body
    * and intercept request from going to wire.
-   * @param credential
    */
-  public createPipeline(
-    credential: SharedKeyCredential | AnonymousCredential | TokenCredential
-  ): Pipeline {
-    const isAnonymousCreds = credential instanceof AnonymousCredential;
-    const policyFactoryLength = 3 + (isAnonymousCreds ? 0 : 1); // [deserilizationPolicy, BatchHeaderFilterPolicyFactory, (Optional)Credential, BatchRequestAssemblePolicyFactory]
-    let factories: RequestPolicyFactory[] = new Array(policyFactoryLength);
-
-    factories[0] = deserializationPolicy(); // Default deserializationPolicy is provided by protocol layer
-    factories[1] = new BatchHeaderFilterPolicyFactory(); // Use batch header filter policy to exclude unnecessary headers
-    if (!isAnonymousCreds) {
-      factories[2] = isTokenCredential(credential)
-        ? bearerTokenAuthenticationPolicy(credential, StorageOAuthScopes)
-        : credential;
-    }
-    factories[policyFactoryLength - 1] = new BatchRequestAssemblePolicyFactory(this); // Use batch assemble policy to assemble request and intercept request from going to wire
-
-    return new Pipeline(factories, {});
+  public createPipelineOptions(): PipelineOptions {
+    return {
+      updatePipelinePolicies: (requestPolicyFactories) => {
```

Comment:
Extend default pipeline policies might be worth discussion:

The original idea here is to use a private internal pipeline assembled for serialization sub requests. 

With limited policies, it easy to control logic correctness.

Default pipeline may contain policies which adds user-agent or do logging and etc. These are not wished for sub request.

Reply:
Good catch! Can definitely pare down the policies here.